### PR TITLE
[BUGFIX release] [Fixes #15492] WorkAround WebKit/JSC JIT WorkARound

### LINKED ISSUE #15492: getOwner(this) is sometimes returning undefined after ember-2.14.0-beta2
This is related to:

https://github.com/intercom/ember-href-to/issues/81
and
https://github.com/Vestorly/torii/issues/349#issuecomment-302141236
and perhaps 
https://github.com/emberjs/ember.js/issues/15468

It appears that sometimes `getOwner(this)` returns `undefined`, and we are having to work around it with `const owner = getOwner(this) || this.container`

### PR DESCRIPTION
**Pull request by stefanpenner**
Affected version: Version 10.0.3 (12602.4.8)
Unaffected versions: Safari 11.0, WebKit 12604.1.29

Which suggests the issue has been addressed upstream.

---

The following commit https://github.com/emberjs/ember.js/commit/986710f9c57a9ff4520034b600e7fea93b56b955#diff-7e13eecefe753df1d82ce67b32bc4366R566 remove an implicit toBoolean conversion in WeakMap_peekParentMeta which apparently trips up Webkit…

Simply reversing the expression addressed the issue.

failed: `if (meta === null || meta !== undefined) { … }`
passed: `if (meta !== undefined || meta === null) { … }`

We also, confirmed that reverting the one line in question to `meta === null || !meta) {` fixed Safari.

This PR removes the need for the an unused feature which avoids the expression entirely.

@krisselden / @stefanpenner

### CODE REVIEW

**Comment by stefanpenner** [comment]
found we still set null explicitly here: https://github.com/emberjs/ember.js/blob/master/packages/ember-runtime/lib/mixins/array.js#L641

File: packages/ember-metal/lib/meta.js [hunk 1]
```diff
@@ -518,7 +518,7 @@ if (HAS_NATIVE_WEAKMAP) {
       // stop if we find a `null` value, since
```
Comment (krisselden) [inline packages/ember-metal/lib/meta.js]: this comment seems incorrect

File: packages/ember-runtime/lib/mixins/array.js [hunk 2]
```diff
@@ -638,7 +638,7 @@ const ArrayMixin = Mixin.create(Enumerable, {
 function EachProxy(content) {
   this._content = content;
   this._keys = undefined;
-  this.__ember_meta__ = null;
+  this.__ember_meta__ = undefined;
```
Comment (krisselden) [inline packages/ember-runtime/lib/mixins/array.js]: Is this worth keeping?

File: packages/ember-metal/lib/meta.js [hunk 3]
```diff
@@ -509,7 +509,7 @@ if (HAS_NATIVE_WEAKMAP) {
   peekMeta = function WeakMap_peekParentMeta(obj) {
     let pointer = obj;
     let meta;
-    while (pointer !== undefined && pointer !== null) {
+    while (pointer !== undefined && null !== pointer) {
```
Comment (krisselden) [inline packages/ember-metal/lib/meta.js]: what is this for? this seems unrelated

File: packages/ember-runtime/lib/mixins/array.js (hunk 2 above)
Comment (stefanpenner) [inline packages/ember-runtime/lib/mixins/array.js]: I think we can strip it once we drop support for old browsers entirely. For now, keeping the shape stable seems important?

File: packages/ember-metal/lib/meta.js (hunk 3 above)
Comment (stefanpenner) [inline packages/ember-metal/lib/meta.js]: interesting...

File: packages/ember-runtime/lib/mixins/array.js (hunk 2 above)
Comment (krisselden) [inline packages/ember-runtime/lib/mixins/array.js]: why not then call meta(this) instead?

Comment (stefanpenner) [inline packages/ember-runtime/lib/mixins/array.js]: I wanted to avoid making any significant changes for bugfix we intend to backport without proper measurement.

But I can gladly make the change

**Review by krisselden — APPROVED** [review]
